source/MaterialXTest/Runtime.cpp

```diff
@@ -654,6 +653,65 @@ TEST_CASE("Runtime: FileIo", "[runtime]")
     }
 }
 
+TEST_CASE("Runtime: FileIo NodeGraph", "[runtime]")
+{
+    mx::FileSearchPath searchPath;
+    searchPath.append(mx::FilePath::getCurrentPath() / mx::FilePath("libraries"));
+ 
+    // Load in stdlib to a stage.
+    mx::RtStage libStage = mx::RtStage::createNew("libs");
+    mx::RtFileIo libFileIo(libStage.getObject());
+    libFileIo.readLibraries({ "stdlib" }, searchPath);
+
+    // Create a main stage referencing the libs stage.
+    mx::RtStage mainStage = mx::RtStage::createNew("main");
+    mainStage.addReference(libStage.getObject());
+
+    // Create a nodegraph.
+    mx::RtNodeGraph graph = mx::RtNodeGraph::createNew(mainStage.getObject(), "testgraph");
+    graph.addPort("in", mx::RtType::FLOAT);
+    graph.addPort("out", mx::RtType::FLOAT, mx::RtPortFlag::OUTPUT);
+    mx::RtPort graphIn = graph.findPort("in");
+    mx::RtPort graphOut = graph.findPort("out");
+    mx::RtPort graphInSocket = graph.findInputSocket("in");
+    mx::RtPort graphOutSocket = graph.findOutputSocket("out");
+    REQUIRE(graphIn);
+    REQUIRE(graphOut);
+    REQUIRE(graphInSocket);
+    REQUIRE(graphOutSocket);
+
+    // Add nodes to the graph.
+    mx::RtNodeDef addNodeDef = mainStage.findElementByName("ND_add_float");
+    mx::RtNode add1 = mx::RtNode::createNew(graph.getObject(), addNodeDef.getObject());
+    mx::RtNode add2 = mx::RtNode::createNew(graph.getObject(), addNodeDef.getObject());
+    mx::RtNode add3 = mx::RtNode::createNew(graph.getObject(), addNodeDef.getObject());
+    mx::RtNode::connect(graphInSocket, add1.findPort("in1"));
+    mx::RtNode::connect(add1.findPort("out"), add2.findPort("in1"));
+    mx::RtNode::connect(add2.findPort("out"), graphOutSocket);
+
+    // Create a node on root level and connect it downstream after the graph.
+    mx::RtNode add4 = mx::RtNode::createNew(mainStage.getObject(), addNodeDef.getObject());
+    mx::RtNode::connect(graphOut, add4.findPort("in1"));
+
+    mx::RtWriteOptions options;
+    options.writeIncludes = false;
+
+    // Save the stage to file for inspection.
+    const mx::FilePath filename = graph.getName().str() + "_export.mtlx";
+    mx::RtFileIo fileIo(mainStage.getObject());
+    fileIo.write(filename, &options);
+
+    // Read the saved file to another stage.
+    mx::RtStage anotherStage = mx::RtStage::createNew("another");
+    anotherStage.addReference(libStage.getObject());
+    fileIo.setObject(anotherStage.getObject());
+    fileIo.read(filename, searchPath);
+
+    // Save the re-read stage to file for inspection.
+    const mx::FilePath filename2 = graph.getName().str() + "_another_export.mtlx";
+    fileIo.write(filename2, &options);
```

Comment:
write -> read -> write, again, to make sure both read and write of the data works correctly.

Reply:
Could be wrong, but shouldn't you be verifying that the files are the same?

Reply:
I could, in this case I was lazy and just used eye balling to compare the files :)

Reply:
I can add an automatic compare.